src/Blockfrost.Api/Services/IPFS/BlockfrostService.Gateway.cs

```diff
@@ -23,112 +22,106 @@ public async Task GatewayAsync(string iPFS_path, CancellationToken cancellationT
+                                    throw new ApiException("Response was null which was not expected.", status, objectResponse.Text, headers, null);
+                                }
+                                throw new ApiException<NotFoundResponse>("Component not found", status, objectResponse.Text, headers, objectResponse.Object, null);
+                            }
+                        case 418:
+                            {
+                                var objectResponse = await ReadObjectResponseAsync<UnsupportedMediaTypeResponse>(response, headers, cancellationToken).ConfigureAwait(false);
+                                if (objectResponse.Object == null)
+                                {
+                                    throw new ApiException("Response was null which was not expected.", status, objectResponse.Text, headers, null);
+                                }
+                                throw new ApiException<UnsupportedMediaTypeResponse>("IP has been auto-banned for extensive sending of requests after usage limit has been reached", status, objectResponse.Text, headers, objectResponse.Object, null);
+                            }
+                        case 429:
+                            {
+                                var objectResponse = await ReadObjectResponseAsync<TooManyRequestsResponse>(response, headers, cancellationToken).ConfigureAwait(false);
+                                if (objectResponse.Object == null)
+                                {
+                                    throw new ApiException("Response was null which was not expected.", status, objectResponse.Text, headers, null);
+                                }
+                                throw new ApiException<TooManyRequestsResponse>("Usage limit reached", status, objectResponse.Text, headers, objectResponse.Object, null);
+                            }
+                        case 500:
+                            {
+                                var objectResponse = await ReadObjectResponseAsync<InternalServerErrorResponse>(response, headers, cancellationToken).ConfigureAwait(false);
+                                if (objectResponse.Object == null)
+                                {
+                                    throw new ApiException("Response was null which was not expected.", status, objectResponse.Text, headers, null);
+                                }
+                                throw new ApiException<InternalServerErrorResponse>("Internal Server Error", status, objectResponse.Text, headers, objectResponse.Object, null);
+                            }
+                        default:
+                            {
+#if NET
+                                var responseData = await response.Content.ReadAsStringAsync(cancellationToken).ConfigureAwait(false);
+#else
+                                var responseData = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+#endif
```

Comment:
This is unavoidable and happens when you support multiple target frameworks. . 
But I think we can keep these compiler instructions to a minimum.